string vocal_output
string data_type
